backends/dpdk/spec.cpp

```diff
@@ -157,12 +169,22 @@ std::ostream &IR::DpdkHeaderType::toSpec(std::ostream &out) const {
 std::ostream &IR::DpdkStructType::toSpec(std::ostream &out) const {
     if (getAnnotations()->getSingle("__packet_data__")) {
         for (auto it = fields.begin(); it != fields.end(); ++it) {
-            if (auto t = (*it)->type->to<IR::Type_Name>())
+            if (auto t = (*it)->type->to<IR::Type_Name>()) {
                 out << "header " << (*it)->name << " instanceof "
                     << t->path->name;
-            else {
-                std::cout << (*it)->type->node_type_name() << std::endl;
-                BUG("Unsupported type");
+            } else if (auto t = (*it)->type->to<IR::Type_Stack>()) {
+                if (!t->elementType->is<IR::Type_Name>())
+                    BUG("%1% Unsupported type", t->elementType);
+                cstring type_name = t->elementType->to<IR::Type_Name>()->path->name;
+                if (!t->size->is<IR::Constant>()) {
```

Comment:
you'd think that dpdk would support dynamic indexes.

Reply:
True, will fix when there is a use case for it.